include/SFML/Graphics/RenderTarget.hpp

```diff
@@ -82,6 +83,29 @@ class SFML_GRAPHICS_API RenderTarget
     ////////////////////////////////////////////////////////////
     void clear(const Color& color = Color(0, 0, 0, 255));
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Clear the stencil buffer to a specific value
+    ///
+    /// The specified value is truncated to the bit width of
+    /// the current stencil buffer.
+    ///
+    /// \param value Stencil value to clear to
+    ///
+    ////////////////////////////////////////////////////////////
+    void clear(int value);
+
+    ////////////////////////////////////////////////////////////
+    /// \brief Clear the entire target with a single color and stencil value
+    ///
+    /// The specified stencil value is truncated to the bit
+    /// width of the current stencil buffer.
+    ///
+    /// \param color Fill color to use to clear the render target
+    /// \param value Stencil value to clear to
+    ///
+    ////////////////////////////////////////////////////////////
+    void clear(const Color& color, int value);
```

Comment:
I guess having all combinations of (color, stencil) is fine for 2 arguments, but this might not scale that well in case we ever want to provide more parameters for `clear()`. Not sure if that's realistic, we can always address such a problem only when we have it (in the worst case by deprecating existing `clear()` overloads).

Reply:
> or should we use std::int32_t here?

https://github.com/SFML/SFML/issues/2325 

I'd generally prefer we err more on the side of sized integers. In practice though `int` is 32 bits on everything but Win16 platforms so in this particular case it's a safe assumption. I'll do the work to move us over to more fixed width integer types if the SFML team can come to agreement on how to use them.

Reply:
`value` depends on the size of the stencil buffer, which itself depends on what is supported by the system and what the user requests during window creation.

`int` was chosen here because that is also what the underlying API uses. I personally also don't know how larger values would be passed if some hypothetical system were to support stencil buffers with more than 32 bits.

Usually even if a system supported more than e.g. 8 bits only 8 bits are requested so as not to waste memory. Typical rendering techniques using the stencil buffer don't need it to be that large.

Using a fixed width type here wouldn't feel right. The default type for numeric literals whether in decimal or hexadecimal notation is `int` and those are the 2 most common ways of clearing the stencil buffer.